bokeh/core/tests/test_properties.py

```diff
@@ -1766,6 +1766,18 @@ class Foo(HasProps):
         with pytest.raises(ValueError):
             f.x = dict(field="foo", units="junk")
 
+def test_dataspec_dict_to_serializable():
+    for typ in (NumberSpec, StringSpec, FontSizeSpec, ColorSpec, DataDistanceSpec, ScreenDistanceSpec):
+        class Foo(HasProps):
+            x = typ("x")
+        foo = Foo(x=dict(field='foo'))
+        props = foo.properties_with_values(include_defaults=False)
+        if typ is DataDistanceSpec:
+            assert props['x']['units'] == 'data'
+        elif typ is ScreenDistanceSpec:
+            assert props['x']['units'] == 'screen'
+        assert props['x']['field'] == 'foo'
+
```

Comment:
Yes, good point, that was the whole point of the change after all. I was only implicitly testing that because it would otherwise raise the ValueError.